M2/Macaulay2/m2/examples.m2

```diff
@@ -169,11 +170,35 @@ extractExamples = docBody -> (
 -- examples: get a list of examples in a documentation node
 -----------------------------------------------------------------------------
 
+-- low-level unexported method called by getExampleOutput
+examples' := method(Dispatch => Thing)
+examples' Hypertext   := stack @@ extractExamplesLoop
+examples' String      := examples' @@ makeDocumentTag
+examples' DocumentTag := tag  -> (
+    rawdoc := fetchAnyRawDocumentation tag;
+    try examples' DIV {rawdoc.Description})
```

Comment:
Sorry, I don't mean to keep dragging out this PR, but I still don't like this for several reasons:
1. naming this `examples'`, when every other internal function in the chain that eventually calls this has a descriptive name seems strange.
2. we don't need `examples' String`, instead insert `makeDocumentTag` in `getExampleOutput` where this is called.
3. `try` here is going to hide legitimate errors that may occur in `extractExamplesLoop` (which should be rare, but not impossible). Let's keep the previous behavior of checking if `rawdoc` is not null and has that key.
4. what happened to the `raise`?

Reply:
No worries -- these are great points!

Reply:
The `raise` previously moved the examples down a line, I suppose so that the `o1 = ` wouldn't muddy up the output?

```m2
i1 : examples rank

o1 = 
     R = ZZ/101[x,y,z]
     p = vars R;
     rank kernel p
     rank cokernel p
     C = res cokernel p
     rank C
```
But with the new behavior, we have comments, which serve a similar purpose:

```m2
i1 : examples rank

o1 = -- examples for tag: rank
     -- ../packages/Macaulay2Doc/functions/rank-doc.m2:32:0
     R = ZZ/101[x,y,z]
     p = vars R;
     rank kernel p
     rank cokernel p
     C = res cokernel p
     rank C
```